frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -804,35 +869,47 @@ mod tests {
 
 	#[test]
 	fn ocw_check_prevent_duplicate() {
+		const OFFCHAIN_REPEAT: BlockNumber = 5;
+
 		let (mut ext, _) = ExtBuilder::default().build_offchainify(0);
 		ext.execute_with(|| {
 			roll_to(25);
 			assert!(MultiPhase::current_phase().is_unsigned());
 
 			// first execution -- okay.
-			assert!(MultiPhase::try_acquire_offchain_lock(25).is_ok());
+			assert!(MultiPhase::try_acquire_offchain_lock(25, OFFCHAIN_REPEAT.into()).is_ok());
 
 			// next block: rejected.
-			assert!(MultiPhase::try_acquire_offchain_lock(26).is_err());
+			assert!(MultiPhase::try_acquire_offchain_lock(26, OFFCHAIN_REPEAT.into()).is_err());
 
 			// allowed after `OFFCHAIN_REPEAT`
-			assert!(MultiPhase::try_acquire_offchain_lock((26 + OFFCHAIN_REPEAT).into()).is_ok());
+			assert!(MultiPhase::try_acquire_offchain_lock(
+				(26 + OFFCHAIN_REPEAT).into(),
+				OFFCHAIN_REPEAT.into()
+			)
+			.is_ok());
 
 			// a fork like situation: re-execute last 3.
-			assert!(
-				MultiPhase::try_acquire_offchain_lock((26 + OFFCHAIN_REPEAT - 3).into()).is_err()
-			);
-			assert!(
-				MultiPhase::try_acquire_offchain_lock((26 + OFFCHAIN_REPEAT - 2).into()).is_err()
-			);
-			assert!(
-				MultiPhase::try_acquire_offchain_lock((26 + OFFCHAIN_REPEAT - 1).into()).is_err()
-			);
+			assert!(MultiPhase::try_acquire_offchain_lock(
+				(26 + OFFCHAIN_REPEAT - 3).into(),
+				OFFCHAIN_REPEAT.into()
+			)
+			.is_err());
+			assert!(MultiPhase::try_acquire_offchain_lock(
+				(26 + OFFCHAIN_REPEAT - 2).into(),
+				OFFCHAIN_REPEAT.into()
+			)
+			.is_err());
+			assert!(MultiPhase::try_acquire_offchain_lock(
+				(26 + OFFCHAIN_REPEAT - 1).into(),
+				OFFCHAIN_REPEAT.into()
+			)
+			.is_err());
 		})
 	}
 
 	#[test]
-	fn ocw_only_runs_when_signed_open_now() {
+	fn ocw_runs_when_signed_open_now() {
```

Comment:
I think for the purpose of this test, we should only ensure that the initial submission will cache it. 

Then, we need a new test that tests: 

1. after OffchainRepeat, we re-submit the cached thing.
2. after OffchainRepeat, if the cache is empty, we re calculate, and submit and re-cache it.

Reply:
449b799331684d3c79f240fe462a8198dfd65f85 and 85f715e847c8e4ad8ab2ea65515cb5c09e90114f